packages/python-runner/src/PythonRunner.ts

```diff
@@ -18,35 +18,49 @@ import { Kernel } from '@jupyterlab/services';
 import { CodeEditor } from '@jupyterlab/codeeditor';
 import { Dialog, showDialog } from '@jupyterlab/apputils';
 
+const RUN_BUTTON_CLASS = 'elyra-PythonEditor-Run';
+
 /**
  * Class: An enhanced Python Script Editor that enables developing and running the script
  */
 export class PythonRunner {
+  id: string;
   kernel: Kernel.IKernel;
   model: CodeEditor.IModel;
   kernelSettings: Kernel.IOptions;
+  startingKernel: boolean;
 
   /**
    * Construct a new runner.
    */
-  constructor(model: CodeEditor.IModel) {
+  constructor(model: CodeEditor.IModel, id: string) {
+    this.id = id;
     this.kernel = null;
     this.model = model;
+    this.startingKernel = false;
   }
 
+  startKernelLock(lock: boolean): void {
```

Comment:
I don't think DOM manipulation on an element defined in the widget should be done in PythonRunner, since it's supposed to only deal with the backend. I was thinking some sort of callback like we did with `handleKernelMsg()` that notifies the UI, so that button lock is done from the widget.

Reply:
That's fair, I moved it in here because I needed to access it from multiple functions, but I realized I would hand it to the constructor as a callback function instead of passing in the id and it would work the same.